backend/src/main/java/com/example/dontwastetomuch/game/GameService.java

```diff
@@ -45,13 +47,14 @@ public void editGame(Game game) {
         }
     }
 
+
     public void addMyGame(MyUser myUser, String gameId) {
         GameData gameData = new GameData(gameId);
-        if (myUser.getGameData().contains(gameData)){
-            throw new IllegalStateException("Game already added to your list");
-        } else {
+        if (myUser.getGameData() == null || !myUser.getGameData().contains(gameData)){
             myUser.addGameData(gameData);
             myUserRepo.save(myUser);
```

Comment:
I would import the `MyUserService` instead of `MyUserRepo`. 
but maybe iam mistaken. pls ask a coach about that. 
Also in your Controller i would not import MyUserService to retrive the `MyUser` object from the username string. 
Instead i would just send the Username as String in the Method paramerters of the method u wanna call. 
And then do the retriving of the `MyUser` object in your service.